### PR TITLE
Organize directory structure of builds repository

### PR DESCRIPTION
**Pull request by YasminBeatriz**
Organize directory structure of builds repository

- Move ``tools/*.py`` to ```lib/subcommands/*.py``` and delete ```tools/``` directory.

- Create new directory ```scripts/``` and move ```validate_*.py``` to ```scripts/```.

- Update Jenkins jobs to use new path ```scripts/validate_*.py``` (in infrastructure repo)

- Move directory ```mock_configs/``` to ```config/mock/```.

- Move ```config.yaml``` from the root to ```config/host_os.yaml```.

### CODE REVIEW

**Comment by ghost** [comment]
Can one of the admins verify this patch?

**Comment by ghost** [comment]
Can one of the admins verify this patch?

**Comment by ghost** [comment]
Can one of the admins verify this patch?

**Review by ghost — CHANGES_REQUESTED** [review]
The paths to mock config files need to be updated in `config/host_os.yaml`

**Comment by ghost** [comment]
ok to test

**Comment by olavphibm** [comment]
start tests

**Review by ghost — DISMISSED** [review]
The build is failing because we need to apply the changes in the infrastructure repo.

Minor: there's no need to repeat the commit short message in the commit longer message.

**Review by ghost — APPROVED** [review]

File: BUILD_OLD_VERSIONS.md [hunk 1]
```diff
@@ -7,20 +7,20 @@ before building:
 Update yum repository to CentOS 7.2:
 
 ```
-$ sed -i 's|http://mirror.centos.org/altarch/|http://vault.centos.org/altarch/7.2.1511/|' mock_configs/CentOS/7/CentOS-7-ppc64le.cfg
+$ sed -i 's|http://mirror.centos.org/altarch/|http://vault.centos.org/altarch/7.2.1511/|' configs/mock/CentOS/7/CentOS-7-ppc64le.cfg
```
Comment (ghost) [inline BUILD_OLD_VERSIONS.md]: Remove the "s" from "configs"

File: BUILD_OLD_VERSIONS.md [hunk 2]
```diff
@@ -7,20 +7,20 @@ before building:
...
 ```
 
 Update distro version to 7.2:
 
 ```
-$ sed -i "s|distro_version:.*|distro_version: \"7.2\"|" config.yaml
-$ sed -i -E "s|  '7': (\"\./mock_configs.*)|  '7.2': \1|" config.yaml
+$ sed -i "s|distro_version:.*|distro_version: \"7.2\"|" config/host_os.yaml
+$ sed -i -E "s|  '7': (\"\./configs/mock.*)|  '7.2': \1|" config/host_os.yaml
```
Comment (ghost) [inline BUILD_OLD_VERSIONS.md]: Remove the "s" from "configs"

**Review by ghost — DISMISSED** [review]

**Review by olavph — APPROVED** [review]

**Pull request merged**